src/Compilers/Core/MSBuildTask/ManagedCompiler.cs

```diff
@@ -494,10 +524,13 @@ public string GenerateResponseFileContents()
         /// <summary>
         /// Get the command line arguments to pass to the compiler.
         /// </summary>
-        private string[] GetArguments(string commandLineCommands, string responseFileCommands)
+        private string[] GetArguments(string commandLineCommands, string responseFileCommands, bool emitServerLog)
```

Comment:
Since there is only one caller of GetArguments that requires logging, please move the logging there.